specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/examples/associatedTenantsCreateOrUpdate.json

```diff
@@ -0,0 +1,51 @@
+{
+  "parameters": {
+    "api-version": "2023-04-01",
+    "billingAccountName": "00000000-0000-0000-0000-000000000000:00000000-0000-0000-0000-000000000000_2019-05-31",
+    "associatedTenantName": "11111111-1111-1111-1111-111111111111",
+    "parameters": {
+      "properties": {
+        "friendlyName": "Contoso Finance",
+        "billingManagementState": "Active",
+        "provisioningState": "Pending"
```

Comment:
Why would the PUT payload include billingManagementState/provisioningState?

Reply:
BillingManagementState and ProvisioningState are not readOnly properties. By setting `billingManagementState` to `Active` and `provisioningState` to `Pending` the API will create the corresponding MCAPI entities behind the scenes.

This behavior is necessary because there are two MCAPI backing entities for a single AssociatedTenant instead of just one.

Reply:
What are the MCAPI backing entities?

Why not simply have an associatedTenant resource for each type of relationship? If tenant X adds tenant Y for both billing management *and* provisioning, then just have 2 associations with different types. That would then also simplify the filtering of associations by a specific type of relationship.

Reply:
We can update it, but we modeled the API based on how we explain these to customers - "You create a relationship with a tenant and enable two settings - Provisioning and Billing Management". Now if the API becomes associated tenant relationships rather than associated tenants then UX and API aren't consistent. I would recommend keeping the UX and the API the same to avoid confusion.

Reply:
I'm fine with keeping it as 1 resource, but then should decouple the *actual state* from the *mode*. Right now we're mixing the two, and it looks inconsistent/complex.

Could we have a "mode" property with values Provisioning, Billing, ProvisioningAndBilling or something like that? Then have a state property (or two state properties if absolutely needed) to describe the actual state...

Reply:
I don't see a problem with adding a read-only "mode" property. But we will need to have two state properties as provisioning tenants have extra states line "pending" and "revoked."

Reply:
That's fine. I think separating intent from state will make this easier to reason over.

Reply:
We'll take notes on this for future versions. For this version we will keep design as is.

Reply:
After syncing with Ram and Justin, we do not want to move forward with the `mode` model for AssociatedTenants. 

Wilco has said offline that if @grzegorzzygmunt signs off then he will not block this design.

Reply:
I agree with Wilco that having two lists of tenants would be a cleaner approach, as I mentioned this during original discussions. But I do not feel strong enough about it to push against using a single associatedTenants list to represent both.
Given that, I am okay with the current approach to have billingManagmentState and provisioningState.